src/wallet/coinselection.cpp

```diff
@@ -93,19 +93,24 @@ std::optional<SelectionResult> SelectCoinsBnB(std::vector<OutputGroup>& utxo_poo
         bool backtrack = false;
         if (curr_value + curr_available_value < selection_target || // Cannot possibly reach target with the amount remaining in the curr_available_value.
             curr_value > selection_target + cost_of_change || // Selected value is out of range, go back and try other branch
-            (curr_waste > best_waste && (utxo_pool.at(0).fee - utxo_pool.at(0).long_term_fee) > 0)) { // Don't select things which we know will be more wasteful if the waste is increasing
+            (utxo_pool.at(0).fee - utxo_pool.at(0).long_term_fee) > 0) { // Don't select more things if the current fee rate is expensive.
             backtrack = true;
         } else if (curr_value >= selection_target) {       // Selected value is within range
-            curr_waste += (curr_value - selection_target); // This is the excess value which is added to the waste for the below comparison
+
+            // This is the excess waste for the current iteration.
+            curr_waste = (curr_value - selection_target);
```

Comment:
This is just the excess not the waste. Check line42 in this file or `GetSelectionWaste` for the waste formula.

Reply:
No, `waste` Contains `excess` but there are other components to it